scripts/safe_template_linter.py

```diff
@@ -1955,6 +1950,291 @@ def _find_mako_expressions(self, mako_template):
         return expressions
 
 
+class PythonLinter(BaseLinter):
+    """
+    The linter for Python files.
+    """
+
+    def __init__(self):
+        """
+        Init method.
+        """
+        super(PythonLinter, self).__init__()
+        self._skip_python_dirs = self._skip_dirs + ('tests',)
+
+    def process_file(self, directory, file_name):
+        """
+        Process file to determine if it is a Python file and
+        if it is safe.
+
+        Arguments:
+            directory (string): The directory of the file to be checked
+            file_name (string): A filename for a potential JavaScript file
+
+        Returns:
+            The file results containing any violations.
+
+        """
+        file_full_path = os.path.normpath(directory + '/' + file_name)
+        results = FileResults(file_full_path)
+
+        if not results.is_file:
+            return results
+
+        if file_name.lower().endswith('.py') is False:
+            return results
+
+        # don't lint this file
+        if file_name == os.path.basename(__file__):
```

Comment:
I'm not sure I understand why we're checking this - what files are we trying to exclude here?